### DISCUSSION TITLE
Aptos Keyless Deeplink Ephemeral keypair Issue

### DISCUSSION THREAD

**Discussion by BigOTF** [aptos-labs/aptos-developer-discussions#556] [Questions] [answered]
### Describe your question in detail.

How can I create and validate an ephemeral key pair using the provided EphemeralKeyPair functions from the Aptos SDK, and what encoding mechanism is used for serialization and deserialization?

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by Pcharlesme** [comment]
T @BigOTF oops, understood .  To create a new ephemeral key pair, the createEphemeralKeyPair function uses the EphemeralKeyPair.generate() method from the Aptos SDK
`const newKeyPair = createEphemeralKeyPair();
console.log(newKeyPair);
`
after then you can Validate an Ephemeral Key Pair
you can use below snippet 

`const keyPair = createEphemeralKeyPair();
if (isValidEphemeralKeyPair(keyPair)) {
  console.log("The key pair is valid!");
} else {
  console.log("The key pair is expired or invalid.");
}`

**Comment by Pcharlesme** [comment]
you can debug it by follow this step  
Use createEphemeralKeyPair to generate a new key pair.
Validate the key pair using isValidEphemeralKeyPair or validateEphemeralKeyPair.
Encode the key pair for storage or transmission using EphemeralKeyPairEncoding.encode.
use this to check the response 

 
```
const encodedKeyPair = EphemeralKeyPairEncoding.encode(newKeyPair);
console.log(encodedKeyPair);

 
const decodedKeyPair = EphemeralKeyPairEncoding.decode(encodedKeyPair);
console.log(decodedKeyPair);

```
Decode the serialized data back into an EphemeralKeyPair using EphemeralKeyPairEncoding.decode

**Comment by Pcharlesme** [comment, marked as answer]
```
export const validateEphemeralKeyPair = (
  keyPair: EphemeralKeyPair

): EphemeralKeyPair | undefined =>
  isValidEphemeralKeyPair(keyPair) ? keyPair : undefined;

export const isValidEphemeralKeyPair = (keyPair: EphemeralKeyPair): boolean => {
  if (keyPair.isExpired()) return false;
  return true;
};
```

**Reply by BigOTF** [reply]
Thanks for your contributions , I followed the guidelines and snippets shared, I was able to resolved the issues  seamlessly
